cvat/apps/tracking/tracker.py

```diff
@@ -83,46 +83,42 @@ def __init__(self, trackerType = "BOOSTING"):
             'MOSSE': cv2.TrackerMOSSE_create,
             'GOTRUN': cv2.TrackerGOTURN_create,
         }
-        if trackerType not in trackerTypes_constructor:
+
+        if trackerType not in self.trackers_constructor:
             raise Exception("Tracker type not known:" + trackerType)
-        self._tracker = trackerTypes_constructor[trackerType]()
 
-    def track_rectangles(self, task, start_shape, stop_frame):
-        """
-        Follow an the rectangle in consecutive frames of a task.
-        :param Task task: The Django Task with the images
-        :param TrackedShape start_shape: Start tracking with this shape; This
-            specifies the frame to start at (start_shape.frame).
-        :param int stop_frame: Stop tracking at this frame (excluded).
-        :return: List of Shapes (Rectangles) with new shapes.
-        """
-        if not isinstance(start_shape, TrackedShape):
-             raise Exception("start_shape must be of type TrackedShape")
+        self._tracker_type = trackerType
 
+    def track_rectangles(self, task, start_shapes, stop_frame):
         # Only track in to future.
-        start_frame = start_shape.frame
+        start_frame = start_shapes[0].frame
         if stop_frame < start_frame:
             return []
 
         # Load the image iterable for range of frames
         # and init the tracker with the bounding box from the user given shape
         images = image_iterable_from_task(task, start_frame, stop_frame)
         img0 = next(images)[1]
-        bbox = rectanlge_to_cv_bbox(start_shape.points)
-        no_error = self._tracker.init(img0, bbox)
+
+        trackers = cv2.MultiTracker_create()
+
+        for shape in start_shapes:
+            bbox = rectangle_to_cv_bbox(shape.points)
+            tracker = self.trackers_constructor[self._tracker_type]()
+            trackers.add(tracker, img0, bbox)
 
         #Generated shapes
         shapes_by_tracking = []
         for frame, img  in images:
```

Comment:
lol whats the difference between frame and img?

Reply:
frame is the frame number and image is, well..., the image...

Reply:
```suggestion
        for frame_number, frame  in frames_by_number:
```